lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -78,6 +78,44 @@ def wrapped_function(self, request, **kwargs):
     return wrapped_function
 
 
+def verify_course_exists_and_in_program(view_func):
+    """
+    Raises:
+        An api error if the course run specified by the `course_key` kwarg
+        in the wrapped function is not part of the curriculum of the program
+        specified by the `program_uuid` kwarg
+
+    Assumes that the program exists
+    """
+    @wraps(view_func)
+    @verify_course_exists
+    def wrapped_function(self, request, **kwargs):
+        """
+        Wraps view function
+        """
+        course_key = CourseKey.from_string(kwargs['course_id'])
+        program_uuid = kwargs['program_uuid']
+        program = get_programs(uuid=program_uuid)
+        active_curricula = (c for c in program['curricula'] if c['is_active'])
+        curriculum = next(active_curricula)
```

Comment:
I'd check if `active_curricula` is empty here (as it is for all non-masters programs) and raise the API error if it is, instead of 500-ing.

Reply:
I think i'd actually want to check if `program` has an empty `curricula` right? Or should I check both situations? That we don't have any curricula _and_ that we don't have an active curricula?

Reply:
Checking `active_curricula` will implicitly check `curricula`. If `curricula` is empty, then `active_curricula` will gracefully be assigned to empty as well.

Reply:
Yeah, my question was poorly worded. I meant more to ask, should we be checking both of those situations? If we have a program with curricula and no active curriculum, should we 500 or 404 ...... I guess probably 404

Reply:
Ah, gotcha. I'm gonna say 404, because that situation could happen if partners try enrolling while Program Coordinators are still setting up the program structure.